### PR TITLE
fix(dsrepl): anticipate and handle nodes leaving the cluster

### PR DESCRIPTION
**Pull request by keynslug**
Fixes [EMQX-12365](https://emqx.atlassian.net/browse/EMQX-12365).

Release version: v5.7

## Summary

Attempt to bring together the DS's view of the cluster with mria/ekka's view and related operations, e.g. leave and force-leave.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] ~~Added property-based tests for code which performs user input validation~~
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [ ] ~~Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket~~
- [x] Schema changes are backward compatible

[EMQX-12365]: https://emqx.atlassian.net/browse/EMQX-12365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl [hunk 1]
```diff
@@ -164,32 +171,55 @@
...
-    Nodes = [node() | nodes()],
     lists:foreach(
         fun(#?NODE_TAB{site = Site, node = Node}) ->
             Status =
-                case lists:member(Node, Nodes) of
-                    true -> up;
-                    false -> down
+                case mria:cluster_status(Node) of
+                    running -> "    up";
+                    stopped -> "(↓) down";
```
Comment (ieQu1) [inline apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl:191]: Not sure if we want unicode in the command output.

**Review by ieQu1 — APPROVED** [review]

**Comment by ieQu1** [comment]
Q: is it possible to display `emqx_ds_replication_layer_meta:replica_set_transitions` in the status?

**Review by ieQu1 — APPROVED** [review]

**Comment by ieQu1** [comment]
The output looks a bit crooked:
```
THIS SITE:
D8894F95DC86DFDB

SITES:
5C6028D6CE9459C7    'emqx@n2.local'        up
D8894F95DC86DFDB    'emqx@n1.local'        up
F4E92DEA197C8EBC    'emqx@n3.local'        up

SHARDS:
Shard                         Replicas
messages/0                        5C6028D6CE9459C7
messages/1                        5C6028D6CE9459C7
messages/10                       5C6028D6CE9459C7
messages/11                       5C6028D6CE9459C7
messages/2                        5C6028D6CE9459C7
messages/3                        5C6028D6CE9459C7
messages/4                        5C6028D6CE9459C7
messages/5                        5C6028D6CE9459C7
messages/6                        5C6028D6CE9459C7
messages/7                        5C6028D6CE9459C7
messages/8                        5C6028D6CE9459C7
messages/9                        5C6028D6CE9459C7
```

**Comment by keynslug** [comment]
> The output looks a bit crooked:

Yeah, that's a consequence of that space is now used for markers denoting down / non-existent sites. 🫠 Will fix in a followup.

**Pull request merged**